### PR TITLE
Controller: cleaning up e2e tests

### PR DESCRIPTION
**Pull request by sbezverk**
Starting to cleanup and adjust e2e tests according to proposed e2e test list. (Controller related tests)

### CODE REVIEW

**Comment by davidz627** [comment]
I think we should wait to merge this until #52 has been resolved (in either direction). These are great e2e tests, but they are being placed alongside unit tests which is not ideal in my opinion.

File: pkg/sanity/controller.go [hunk 1]
```diff
@@ -254,6 +253,154 @@ var _ = Describe("CreateVolume [Controller Server]", func() {
 			})
 		Expect(err).NotTo(HaveOccurred())
 	})
+	It("should not fail when requesting to create a volume with already exisiting name and same capacity.", func() {
```
Comment (lpabon) [inline pkg/sanity/controller.go]: This looks good, but the only thing is that an error may be returned if the driver does not support `RequiredBytes`, like NFS. You will want to check the error returned to be either `OUT_OF_RANGE` or `UNIMPLEMENTED`.

Comment (sbezverk) [reply pkg/sanity/controller.go]: @lpabon thanks a lot for your review, I will address it.

Comment (sbezverk) [reply pkg/sanity/controller.go]: done

File: pkg/sanity/controller.go [hunk 2]
```diff
@@ -240,12 +239,20 @@ var _ = Describe("CreateVolume [Controller Server]", func() {
...
 				},
 			})
-		Expect(err).NotTo(HaveOccurred())
-		Expect(vol).NotTo(BeNil())
-		Expect(vol.GetVolume()).NotTo(BeNil())
-		Expect(vol.GetVolume().GetId()).NotTo(BeEmpty())
-		Expect(vol.GetVolume().GetCapacityBytes()).To(Equal(size))
+		if serverError, ok := status.FromError(err); ok {
+			if serverError.Code() == codes.OutOfRange || serverError.Code() == codes.Unimplemented {
+				Skip("Required bytes not supported")
```
Comment (lpabon) [inline pkg/sanity/controller.go]: 👍

**Pull request merged**